### ISSUE TITLE
Metamask shows large transfer made, but etherscan shows (correctly) smaller transfer to another address

### ISSUE THREAD

**Issue by nlwolf** [MetaMask/metamask-extension#11588]
1. I sent $100 USDC to my Celsius account. Everything went well.
2. I sent remaining balance X to Celsius. TX failed (maybe fee was too small? used 'normal fee' level)
3. Changed my mind, and decided to send remaining balance to FTX instead.
4. Sent $100 test TX to FTX. It shows on my FTX account

But metamask says I sent all my balance to Celsius instead. When i click on the 'transaction confirmed' in the activity log,
it takes me to the $100 FTX transaction instead.

![image](https://user-images.githubusercontent.com/7559234/126646775-3b110c8b-989b-4542-91fd-3ad30a897da2.png)

OS: Win 10
Browser: Chrome Version 91.0.4472.164 (Official Build) (64-bit)
Metamask 9.8.1
Hardware Walled: Trezor Model One 2.3.6

So, in short: metamask incorrectly shows TX of 100 USDC to FTX as full balance TX to Celsius

**Comment by stevecarrol** [comment]
I had a similar issue,
